src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class Ball {
+    private String color;
+    private int number;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public int getNumber() {
+        return number;
+    }
+
+    public void setNumber(int number) {
+        this.number = number;
+    }
+
+    public void setColor() {
+        color = colorSupplier.getRandomColor();
+    }
```

Comment:
The setColor method should accept a parameter to set the ball's color. Currently, it's tightly coupled with the ColorSupplier class. A better approach would be to pass the color as a parameter, which would make the method more flexible and testable.